sn_client/src/register.rs

```diff
@@ -292,17 +303,58 @@ impl ClientRegister {
     }
 
     // Retrieve a `Register` from the closest peers.
-    async fn get_register(client: &Client, name: XorName, tag: u64) -> Result<Register> {
+    async fn get_register_from_peers(client: &Client, name: XorName, tag: u64) -> Result<Register> {
         let address = RegisterAddress { name, tag };
         debug!("Retrieving Register from: {address:?}");
-        let request = Request::Query(Query::Register(RegisterQuery::Get(address)));
+        let request = Request::Query(Query::GetRegister(address));
         let responses = client.send_to_closest(request).await?;
 
-        // We will return the first register we get.
-        for resp in responses.iter().flatten() {
-            if let Response::Query(QueryResponse::GetRegister(Ok(register))) = resp {
-                return Ok(register.clone());
-            };
+        // collect non-failed responses and filter out invalid responses
+        let regs = responses
+            .iter()
+            .flatten()
+            .filter_map(|res| {
+                if let Response::Query(QueryResponse::GetRegister(Ok(register))) = res {
+                    Some(register.clone())
+                } else {
+                    debug!("Got failed response while retrieving Register at {address:?}: {res:?}");
+                    None
+                }
+            })
+            .filter(|reg| {
+                if reg.address() != &address {
+                    debug!(
+                        "Ignoring Register at {address:?} with invalid address",
+                        address = reg.address()
+                    );
+                    return false;
+                }
+                match reg.verify() {
+                    Ok(_) => true,
+                    Err(err) => {
+                        debug!("Ignoring invalid Register at {address:?}: {err:?}");
+                        false
+                    }
+                }
+            })
+            .collect::<Vec<SignedRegister>>();
+
+        debug!(
+            "Got {}/{} valid Register responses for Register at {address:?}",
+            regs.len(),
+            responses.len()
+        );
```

Comment:
For now, even if only 1 node responded with a valid response we take it. Maybe we want to be stricter and require majority?